source/binding/Expression.cpp

```diff
@@ -656,6 +656,17 @@ Expression& Expression::bindName(Compilation& compilation, const NameSyntax& syn
         return *compilation.emplace<DataTypeExpression>(resultType, syntax.sourceRange());
     }
 
+    // Recursive function call
+    if (invocation && symbol->kind == SymbolKind::Variable && result.selectors.empty()) {
+        auto scope = symbol->getParentScope();
+        if (scope && scope->asSymbol().kind == SymbolKind::Subroutine &&
```

Comment:
It's worth storing the result of asSymbol() since it isn't completely free.

Reply:
Nevermind, I was thinking of as<Scope>() which requires checking the kind dynamically. asSymbol() on scope just returns a member directly, so it's basically free.